### ISSUE TITLE
[WC 8.0 Beta]: UI looks disorganised in the cart and checkout pages

### ISSUE THREAD

**Issue by alopezari** [woocommerce/woocommerce-gutenberg-products-block#10357]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In WooCommerce 8.0 beta the style of the items in the cart page looks worse than in previous versions, just like if the layout properties were not consistent in the UI. Examples:

- In WooCommerce 7.9 (correct):
![image](https://github.com/woocommerce/woocommerce/assets/45979455/46745f04-cc5b-4547-a5ee-0486b075d28b)

- In WooCommerce 8.0 beta (wrong)
![image](https://github.com/woocommerce/woocommerce/assets/45979455/38dbd407-7664-4780-82e5-9248e039481b)

As you can see, the `Proceed to checkout` button has significant padding on the upper side and the elements seem closer together. In any case, this change wasn't expected so we can consider it a bug that should be fixed before releasing the official WooCommerce 8.0 version.

### Expected behavior

The UI in the cart and checkout pages should look like in WooCommerce 7.9:
![image](https://github.com/woocommerce/woocommerce/assets/45979455/46745f04-cc5b-4547-a5ee-0486b075d28b)

**NOTE:** After understanding the root of the issue, it would be good to investigate if it affects other pages apart than cart and checkout.

### Actual behavior

The UI in the cart and checkout pages are different and look worse than in previous versions:
![image](https://github.com/woocommerce/woocommerce/assets/45979455/38dbd407-7664-4780-82e5-9248e039481b)

### Steps to reproduce

1. Make sure you are using WooCommerce 8.0 beta.
2. As an admin, add some products to the store.
3. As a shopper, add any of these products to the cart.
4. Open the cart and analyze the UI.
5. Proceed to the checkout page and analyze the UI.

### WordPress Environment

```
WordPress address (URL): https://testsite.com
Site address (URL): https://testsite.com
WC Version: 8.0.0
REST API Version: ✔ 8.0.0
WC Blocks Version: ✔ 10.6.1
Action Scheduler Version: ✔ 3.6.1
Log Directory Writable: ✔
WP Version: 6.2.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.57 (Unix) OpenSSL/1.0.2g
PHP Version: 8.0.29
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.33-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.0.0
WC Database Prefix: wp_
Total Database Size: 5.31MB
Database Data Size: 3.61MB
Database Index Size: 1.70MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.48MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 11
page: 7
post: 2
product: 9
shop_order: 4
wp_template: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

Companion Plugin: by Osk – 1.29
Jetpack: by Automattic – 12.3
WooCommerce: by Automattic – 8.1.0-dev

### Inactive Plugins (2) ###

Akismet Anti-Spam: Spam Protection: by Automattic - Anti Spam Team – 5.1
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### WC Pages ###

Shop base: woocommerce/woocommerce#7 - /shop/
Cart: woocommerce/woocommerce#8 - /cart/
Checkout: woocommerce/woocommerce#9 - /checkout/
My account: woocommerce/woocommerce#10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Three
Version: 1.1
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
product-variation-management
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2023-07-26 06:18:36 +00:00
Options: ✔
Notes: 64
Onboarding: skipped

### Action Scheduler ###

Complete: 16
Oldest: 2023-07-25 06:20:37 +0000
Newest: 2023-07-25 08:14:42 +0000

Failed: 1
Oldest: 2023-07-25 06:21:42 +0000
Newest: 2023-07-25 06:21:42 +0000

Pending: 1
Oldest: 2023-07-26 06:20:37 +0000
Newest: 2023-07-26 06:20:37 +0000


### Status report information ###

Generated at: 2023-07-25 08:59:26 +00:00
```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mikejolley** [comment]
I just tested to beta, and I can see its affected by a separate bug that was patched earlier today:

https://github.com/woocommerce/woocommerce-blocks/pull/10359

In the current beta that means you cannot actually **see** the shortcode cart (from your screenshots), because its redirected to the new block template. With the fix in #10359, shortcode carts on pages will be accessible again.

To reproduce the issue I had to edit the template (Appearance > Edit > Cart), remove the block, and insert the shortcode. The visual issue affects block themes with this setup.

**Comment by mikejolley** [comment]
Seems to be related to shortcodes having P and BR tags added. Related:

- https://core.trac.wordpress.org/ticket/58333
- https://core.trac.wordpress.org/ticket/58366
- https://wordpress.org/support/topic/wordpress-v6-2-1-breaks-the-shortcode-block-in-templates/page/5/
- https://wptavern.com/wordpress-6-2-1-update-breaks-shortcode-support-in-block-templates
